Question 1: values added: 20

Question 2: final result: 20

Question 3: values added:  20

Question 4: The codes return an error because the variable 'result' is defined as 'let' variable type, which means it is accessible within the block scope, so when we call it outside the function, the varible does not exist

Question 5: The codes return an error because the variable 'result' is defined as 'const' variable type, which means it cannot be reassigned after declaration, however, on the line 7, 'result' is reassigned, which causes an error

Question 6: The codes return an error because the line 7 has thrown the error, moreover, the variable 'result' is defined as 'const' variable type, which means it is accessible within the block scope, so when we call it outside the function, the varible does not exist 